src/lv_core/lv_obj.h

```diff
@@ -189,6 +196,7 @@ typedef struct _lv_obj_t
     /*Attributes and states*/
     uint8_t click : 1;       /*1: Can be pressed by an input device*/
     uint8_t drag : 1;        /*1: Enable the dragging*/
+    lv_drag_direction_t drag_dir; /* Which directions the object can be dragged in */
```

Comment:
It will use 1 byte, won't it? `lv_drag_direction_t drag_dir  :2` should be used instead.